### PR TITLE
Random Contributor update

### PR DESCRIPTION
**Pull request by zigelnik**
## Proposed Changes and Benefits

Whenever the 'Show' button is clicked in the header section, a different contributor will be displayed, instead of having to manually refresh the page to see a new contributor.

## Notes to Reviewers

There is a delay of 750 miliseconds that happen when the 'Hide' button is clicked. If the delay is smaller than 750 miliseconds, you would notice the contributor icon changing before the toggling is closed.

### Checklist

- [ X ] I have thoroughly read and understand [The Contribution Guidelines](https://github.com/BeforeIDieCode/BeforeIDieAchievements/blob/main/CONTRIBUTION-GUIDELINES.md).
- [ X ] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behavior.
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc.
  - `Fix` - bug fixes.
  - `Docs` - fixing typos and adding new content.
- [ X ] I have listed my proposed changes and their benefits.
- [ X ] I have linked the corresponding issue in the Issues section.

### CODE REVIEW

File: src/Components/randomcontributor/RandomContributors.jsx [hunk 1]
```diff
@@ -1,29 +1,31 @@
...
-    setIsShown((prev) => !prev);
+  const pickRandomContributor = () => {
+    let randomIndex;
+    do {
+      randomIndex = Math.floor(Math.random() * contributorsData.length);
+    } while (randomIndex === currentIndex);
+    return randomIndex;
   };
 
-  useEffect(() => {
```
Comment (bellsofaba) [inline src/Components/randomcontributor/RandomContributors.jsx:12]: Interesting.

File: src/Components/randomcontributor/RandomContributors.jsx [hunk 2]
```diff
@@ -1,29 +1,31 @@
...
-
-    return () => {
-      clearInterval(interval);
-    };
-  }, []);
+  const handleIsShown = () => {
+    if (isShown) { 
+      const newIndex = pickRandomContributor();
+      setIsShown(false);
+      setTimeout(() => setCurrentIndex(newIndex), 750); // Pick new contributor after 0.75 seconds
```
Comment (XanderRubio) [inline src/Components/randomcontributor/RandomContributors.jsx:21]: I see and makes sense to define a function handleIsShown that will toggle the isShown state and then updates the currentIndex after a 0.75-second delay when hiding the contributor's information.

**Review by XanderRubio — APPROVED** [review]
Hello @zigelnik,

Thank you for taking the initiative to open a new issue and suggesting a UI enhancement. Your idea of displaying the change of the contributor when clicking the show and hide button is reasonable, and I will merge your contribution now.

I would also appreciate your input on improving the UI for the random contributor component. Additionally, any ideas you have for enhancing the appearance would be greatly welcomed. Thank you for your contribution to the UI enhancement, and I will proceed with merging it now.

**Comment by XanderRubio** [comment]
Additionally in the future please open up an issue first before creating a pull request as it is for organizational purposes additionally, that way, we can identify if it is an issue that needs enhancing or fixing (generally speaking, it will always be a yes as a needed enhancement and added feature😉). Thanks, Tal and love the enthusiasm and initiative in looking to enhance the code and UX!

**Comment by zigelnik** [comment]
> Additionally in the future please open up an issue first before creating a pull request as it is for organizational purposes additionally, that way, we can identify if it is an issue that needs enhancing or fixing (generally speaking, it will always be a yes as a needed enhancement and added feature😉). Thanks, Tal and love the enthusiasm and initiative in looking to enhance the code and UX!

Sorry and thanks! 
Now I know how things work.
I will open an Issue first next in the future.

**Comment by XanderRubio** [comment]
> > Additionally in the future please open up an issue first before creating a pull request as it is for organizational purposes additionally, that way, we can identify if it is an issue that needs enhancing or fixing (generally speaking, it will always be a yes as a needed enhancement and added feature😉). Thanks, Tal and love the enthusiasm and initiative in looking to enhance the code and UX!
> 
> Sorry and thanks! Now I know how things work. I will open an Issue first next in the future.

Great and no need to say sorry😇 I appreciate you taking your valuable time to contribute. Enjoy your Sunday!

**Comment by XanderRubio** [comment]
> > > Additionally in the future please open up an issue first before creating a pull request as it is for organizational purposes additionally, that way, we can identify if it is an issue that needs enhancing or fixing (generally speaking, it will always be a yes as a needed enhancement and added feature😉). Thanks, Tal and love the enthusiasm and initiative in looking to enhance the code and UX!
> > 
> > 
> > Sorry and thanks! Now I know how things work. I will open an Issue first next in the future.
> 
> Great and no need to say sorry😇 I appreciate you taking your valuable time to contribute. Enjoy your Sunday!

**Pull request merged**